### ISSUE TITLE
No typechecking when saving twice

### ISSUE THREAD

**Issue by v-moreau** [fizruk/vscode-rzk#70]
Dear maintainers,

I am a bit confused regarding `vscode-rzk` as it seems that, in an ill-typed file, saving once does typecheck and shows the errors whereas saving another time will make the errors disappear. This behavior appears whether or not the file has been changed between the two saves.

For instance, I have a minimal file `test.rzk` whose content is
```rzk
#lang rzk-1

#def weird (A : U)
  : A
  := \ x → x
```
together with a `rzk.yaml` file in the same directory containing
```yaml
include:
  - 'test.rzk'
```
When I save the file once, `vscode-rzk` underlines the definition of `weird` with some red as follows:
![underline](https://github.com/rzk-lang/vscode-rzk/assets/10177882/9075d9ce-df58-4cd2-bf6c-9e36a868975d)
When I put the mouse on it, I get the typechecking error:
![error](https://github.com/rzk-lang/vscode-rzk/assets/10177882/b1f3e2bb-0861-401f-8053-09a858eb2216)
However, when I save the file another time, then the error is gone:
![nothing](https://github.com/rzk-lang/vscode-rzk/assets/10177882/ec3b1e24-1703-4fa2-bce2-f9453e0576d7)

I have tried with a new vscode profile whose sole installed extension is `vscode-rzk`, but I have the same behavior. Is there something wrong on my side, or is it supposed to be how the `vscode-rzk` extension works? For the moment, I thus prefer to use the `rzk typecheck` command in a terminal window to check my file, as I don't see the errors with the vscode extension.

**Comment by fizruk** [comment]
Thanks for the bug report! This is definitely not intended.

I think this has to do with the incorrect caching of the type errors in the Rzk Language Server, which should hopefully be fixed when we merge this PR: https://github.com/rzk-lang/rzk/pull/167

We'll try to reproduce, fix, and release an update sometime this week.

**Comment by fizruk** [comment]
I believe what's happened is on the second save the diagnostics were cleaned, but since nothing changed, we should've preserved the diagnostics (or restored them). I have a very simple fix (https://github.com/rzk-lang/rzk/pull/172) that seems to be working locally for me, the problem should go away with the next release of Rzk (hopefully very soon).

**Comment by fizruk** [comment]
@v-moreau I have just released [`rzk` v0.7.4](https://github.com/rzk-lang/rzk/releases/tag/v0.7.4) that contains the fix to the language server, which should fix this issue for you (reload VS Code so that it prompts you to update Rzk). Please let me know if you experience any further issues!

**Comment by v-moreau** [comment]
Indeed, the bug is now gone on my side too. Thank you very much!
